src/rtk/features/singleRights/singleRightsSlice.ts

```diff
@@ -185,47 +194,92 @@ const singleRightSlice = createSlice({
   },
   extraReducers: (builder) => {
     builder
+      .addCase(delegationAccessCheck.pending, (state, action) => {
+        const serviceID = action.meta.arg.serviceResource.identifier;
+        let nextServicesWithStatus;
+
+        if (
+          state.servicesWithStatus.some(
+            (s: ServiceWithStatus) => s.service?.identifier === serviceID,
+          )
+        ) {
+          nextServicesWithStatus = state.servicesWithStatus.map((sws: ServiceWithStatus) => {
+            if (sws.service?.identifier === serviceID) {
+              sws.isLoading = true;
+            }
+            return sws;
+          });
+        } else {
+          const newServiceWithStatus: ServiceWithStatus = {
+            service: action.meta.arg.serviceResource,
+            isLoading: true,
+          };
+          nextServicesWithStatus = [...state.servicesWithStatus, newServiceWithStatus];
+        }
+
+        state.servicesWithStatus = nextServicesWithStatus;
+      })
+
```

Comment:
Ikke relatert til denne linja men funker koden også for når backend gir http-error eller må man legge til noe kode i BFF-en for å få dette til å funke?

Reply:
Jeg tror det skal fungere med backend sine svar også for jeg har testet med backend kjørende og ved bruk av resurser fra AT.

![image](https://github.com/Altinn/altinn-access-management-frontend/assets/22012997/baedc9f3-caea-444d-91b2-f045546505ae)

I dette bildet er 'Ressurs - OED k6 test' en resurs som jeg har lagt inn policy til i blob-storage, og som derfor gir et suksessfult svar tilbake. 'test_after_merge' har derimot ingen policyfil i blob så når backend ikke finner den (som ikke burde skje i et faktisk miljø og som jeg derfor regner som en unexpected error) så vil backend da sende en Validation Error tilbake, som kommer i form av en 500 Internal Server Error.
Og det plukkes opp i rejected-casen til redux.